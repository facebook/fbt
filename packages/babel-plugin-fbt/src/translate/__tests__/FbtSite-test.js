/**
 * Copyright 2004-present Facebook. All Rights Reserved.
 *
 * @emails oncall+i18n_fbt_js
 * @format
 * @flow strict-local
 */

jest.autoMockOff();

const {FbtSite} = require('../FbtSite');

describe('Test serialization', () => {
  it('should serialize/deserialize as expected', () => {
    const f = new FbtSite(
      {
        'gVKMc/8jq5vnYR5v2bb32g==': {
          text: '{name} has shared {=[number] photos} with you',
          desc: 'example 1',
        },
        'PqPPir8Kg9xSlqdednPFOg==': {
          text: '{name} has shared {=a photo} with you',
          desc: 'example 1',
        },
      },
      {
        t: {
          '*': {
            '*': {
              desc: 'example 1',
              text: '{name} has shared {=[number] photos} with you',
              tokenAliases: {
                '=[number] photos': '=m2',
              },
            },
            _1: {
              desc: 'example 1',
              text: '{name} has shared {=a photo} with you',
              tokenAliases: {
                '=a photo': '=m2',
              },
            },
          },
        },
        m: [
          {
            token: 'name',
            type: 1,
          },
          {
            token: 'number',
            type: 2,
            singular: true,
          },
        ],
      },
      'fbt-demo-project',
    );
    const original = f.serialize();
    const hydrated = FbtSite.deserialize(original).serialize();
    expect(original).toEqual(hydrated);
    expect(original).toMatchSnapshot();
  });
});
