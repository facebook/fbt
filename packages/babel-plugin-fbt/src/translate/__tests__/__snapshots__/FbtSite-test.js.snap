// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test serialization should serialize/deserialize as expected 1`] = `
Object {
  "_d": Object {
    "m": Array [
      Object {
        "token": "name",
        "type": 1,
      },
      Object {
        "token": "number",
        "type": 2,
      },
    ],
    "t": Object {
      "*": Object {
        "*": Object {
          "desc": "example 1",
          "text": "{name} has shared {=[number] photos} with you",
          "tokenAliases": Object {
            "=[number] photos": "=m2",
          },
        },
        "_1": Object {
          "desc": "example 1",
          "text": "{name} has shared {=a photo} with you",
          "tokenAliases": Object {
            "=a photo": "=m2",
          },
        },
      },
    },
  },
  "h2t": Object {
    "PqPPir8Kg9xSlqdednPFOg==": Object {
      "desc": "example 1",
      "text": "{name} has shared {=a photo} with you",
    },
    "gVKMc/8jq5vnYR5v2bb32g==": Object {
      "desc": "example 1",
      "text": "{name} has shared {=[number] photos} with you",
    },
  },
  "p": "fbt-demo-project",
}
`;
